src/main.rs

```diff
@@ -262,42 +268,178 @@ impl SampleTime {
+
+    match with_subprocesses {
+        false => {
+            // Start a single recorder thread
+            let done = done.clone();
+            std::thread::spawn(move || {
+                let result = record(
+                    pid,
+                    sample_rate,
+                    maybe_stop_time,
+                    done,
+                    trace_sender
+                    );
+                error_sender.send(result).unwrap();
+                drop(error_sender);
+            });
+        },
+        true => {
+            // Start a thread which watches for new descendents and starts new recorders when they
+            // appear
+            let done_clone = done.clone();
+            let done = done.clone();
+            std::thread::spawn(move || {
+                let mut pids: HashSet<pid_t> = HashSet::new();
+                let done = done.clone();
+                // we need to exit this loop when the process we're monitoring exits, otherwise the
+                // sender channels won't get closed and rbspy will hang. So we check the done mutex
+                // and call `process_is_running`.
+                while !done_clone.load(Ordering::Relaxed) && process_is_running(pid) {
+                    let descendents = descendents_of(pid).expect("Error finding descendents of pid");
+                    for pid in descendents {
+                        if pids.contains(&pid) {
+                            // already recording it, no need to start a new recording thread
+                            continue;
+                        }
+                        pids.insert(pid);
+                        let trace_sender = trace_sender.clone();
+                        let error_sender = error_sender.clone();
+                        let done = done.clone();
+                        std::thread::spawn(move || {
+                            let result = record(
+                                pid,
+                                sample_rate,
+                                maybe_stop_time,
+                                done,
+                                trace_sender
+                                );
+                            error_sender.send(result).expect("couldn't send error");
+                            drop(error_sender);
+                        });
+                    }
+                    std::thread::sleep(Duration::from_secs(1));
+                }
+            });
+        }
+    }
+    Ok((trace_receiver, result_receiver))
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_spawn_record_children_subprocesses() {
+    // We only run this test on linux because the subprocess code doesn't work on Mac yet
+    let mut process = std::process::Command::new("/usr/bin/ruby").arg("ci/ruby-programs/ruby_forks.rb").spawn().unwrap();
+    let pid = process.id() as pid_t;
+    let (_trace_receiver, result_receiver) = spawn_recorder_children(pid, true, 50, None).unwrap();
```

Comment:
`test test_spawn_record_children_subprocesses has been running for over 60 seconds`

This test isn't terminating on travis. Given we are passing `None` for `maybe_stop_time`, I assume we are not able to detect that the subcommand has exited and thus we are still blocking on the child thread that is error on collecting stack traces?